scripts/test.sh

```diff
@@ -44,16 +48,33 @@ stop_server() {
   fi
 }
 
+startedSimulator=false
+log_temp=
+test_temp_dir=
 cleanup() {
   # Kill started object server
-  stop_server
-
+  stop_server || true
+  
+  # Give the simulator a chance to go down gracefully
+  if $startedSimulator; then
+    osascript -e 'tell app "Simulator" to quit without saving' || true
+    sleep 0.25 # otherwise the pkill following will get it
+  fi
+  
   # Kill all other child processes.
   pkill -P $$ || true
 
   # Kill react native packager
   pkill node || true
   rm -f "$PACKAGER_OUT" "$LOGCAT_OUT"
+  
+  # Cleanup temp files
+  if [ -n "$log_temp" ] && [ -e "$log_temp" ]; then
+    rm "$log_temp" || true
```

Comment:
True if the file is missing, but not if there is an I/O or permission error. Since we are in cleanup I would rather not fail things for those reasons. You are correct and the `-e` check is superfluous, and more a habit from other languages (Python in this case).